lib/buffer.js

```diff
@@ -158,18 +157,16 @@ function assertSize(size) {
  **/
 Buffer.alloc = function(size, fill, encoding) {
   assertSize(size);
-  if (size <= 0)
-    return createBuffer(size);
-  if (fill !== undefined) {
+  if (size > 0 && fill !== undefined) {
```

Comment:
Does a separate check for `0` make sense here, i.e. `size > 0 && fill !== undefined && fill !== 0`?

`Buffer.alloc(size, 0)` is equivalent to `Buffer.alloc(size)`, so just `new FastBuffer(size)` should work faster in that case.